### DISCUSSION TITLE
CodeMagic bundle fails - local bundle runs

### DISCUSSION THREAD

**Discussion by cn-cit** [codemagic-ci-cd/codemagic-docs#1656] [Q&A] [answered]
Hi, if I use CodeMagic to bundleRelease and publish to google play then the app crashes immediately on startup.  However, if I do either of the following, it works fine.
1. local bundleRelease and local copy to my phone - works fine
2. local bundleRelease, upload to google play, publish the release, download the app - works fine

Any clues as to why the CodeMagic bundle will be crashing immediately when the other bundles are fine?  All are using the same source code branch.

**Comment by rafeysup93** [comment]
Hi there,

I will certainly assist you with this. May I please have the build ID to see why it is failing?

**Reply by cn-cit** [reply]
Hi, the build does not fail.  It completes and uploads to google play successfully.  It is when I install from google play onto my phone that the app immediately crashes.  If i do the same bundleRelease on my own laptop and manually upload that to google play and release it, that version installs and works correctly.  But if it helps, the build ID is 64621e1e3b2c3ff9c9d8c561.  Thanks.

**Reply by rafeysup93** [reply]
Are you downloading artifacts from Codemagic build on your laptop for your manual upload to Google Play?

**Reply by cn-cit** [reply]
No.  The artefact I have on my laptop is created using this
`call gradlew bundleRelease`
This produces "app-release.aab" which I upload to google play and then publish the release and install the app onto my phone.  It works fine and does not crash.
N.B. It uses the same keystore locally as I use on CodeMagic.

**Reply by rafeysup93** [reply]
Thank you for confirming that. Could you try to generate an aab bundle file on Codemagic? Download it and upload it manually to Google Play. This way, we will know if the issue is with the generated bundle file or the publishing to Google Play.

**Reply by cn-cit** [reply]
OK, this is what I have done:
1.a. use CodeMagic to bundle my app.
1.b do NOT let CodeMagic upload to GooglePlay
1.c download app-release.aab from CodeMagic to my local environment
1.d upload the CodeMagic bundle from my local environment to Google Play
1.e release on Google Play (internal test track)
1.f install on my phone
::: result = app crashes on startup every time

2.a use my local environment to create app-release.aab
2.b upload my own bundle from my local environment to Google Play
2.c release on Google Play (internal test track)
2.d install on my phone
::: result = app works correctly

**Reply by cn-cit** [reply]
@rafeysup93 any further clues on this?

**Reply by rafeysup93** [reply]
Thank you for waiting patiently. I can see that you are trying to create a React Native Expo application. I am trying to reproduce this issue on our end. Meanwhile, please confirm if you have followed the steps as per this [document](https://docs.codemagic.io/yaml-quick-start/building-a-react-native-app/).

**Reply by cn-cit** [reply]
Hi, I do not actually have an Expo application and it has never been one.  It is a React Native cli app initialized with `npx react-native@latest init AwesomeProject`
I do use Expo modules and they have been installed following all the steps as described in the Expo documentation.  As I mentioned, this all works fine on my local development environment whether I am in debug or release mode.  It also works if I bundleRelease locally and copy directly to my phone or upload that "app-release.aab" to Google Play and download from there.
I have only just built a new bundle from CodeMagic and uploaded to Google Play.  It now works.  I made 2 possible changes that could have resolved the problem.  I am doing some investigation to determine which was the solution.  I will post again when I have some more information.
Thanks.

**Reply by cn-cit** [reply, marked as answer]
OK, I have found the cause of the problem.  logcat gave me a clue of "Cause: null pointer dereference".  I had this once before when setting up functionality around
`globalThis`
which I had removed but recently put back in.  So I took all that out again.  The app still crashed.  logcat then helped with
`#00 pc 000000000006b820  /data/app/~~9Qw2J1rjIf2uV4Dt35V4Qg==/com.phudl.mobile-5N7SsqASbFmJN7NwovBSyQ==/split_config.arm64_v8a.apk!libreanimated.so (offset 0xa23000)`
Well, how was I going to debug anything in reanimated??!!
After a lot of digging I realized my "babel.config.js" with
`  plugins: [
    'react-native-reanimated/plugin',
  ],`
had not been committed to git. DOH!  So, making sure that was committed and then rebuilding with CodeMagic, deploying to Google Play, and installing from there, TADA!!!  The app now works.
All my fault.
However, I sort of thought that not having the babel config in place might have caused the build to fail.  Never mind.
@rafeysup93 thanks for your assistance.
